packages/p2p/src/components/order-details/order-details.jsx

```diff
@@ -164,9 +161,19 @@ const OrderDetails = observer(({ onPageReturn }) => {
                         {is_active_order &&
                             (order_store?.has_order_payment_method_details ? (
                                 <div className='order-details-card--padding'>
-                                    <Text size='xs' weight='bold'>
-                                        {labels.payment_details}
-                                    </Text>
+                                    <section className='order-details-card__title'>
+                                        <Text size='xs' weight='bold'>
+                                            {labels.payment_details}
+                                        </Text>
+                                        <Button
+                                            className='p2p-my-ads__expand-button'
+                                            onClick={() => setShouldExpandAll(!should_expand_all)}
```

Comment:
```suggestion
                                            onClick={() => setShouldExpandAll((prev) => !prev)}
```
If you're having state update issue, you can use the previous state callback: https://reactjs.org/docs/hooks-reference.html#usestate

Reply:
Will incorporate this